pkg/operator/controller/controller.go

```diff
@@ -136,10 +156,38 @@ func (r *ReconcileCDI) Reconcile(request reconcile.Request) (reconcile.Result, e
 
 	// mid delete
 	if cr.DeletionTimestamp != nil {
+		if r.isUpgrading(cr) {
+			//CDI is marked to be deleteted while upgrade flow is in process
+			reqLogger.Info("Deleting  CDI during upgrade")
+			//clenaup all resources of previous installation first
+			err := r.cleanupUnusedResources(reqLogger, cr)
```

Comment:
hmmm, actually is this necessary if the resources are owned by the CDI CR?

Reply:
invokeDeleteCDICallbacks is based on getAllResources, that is if resource is not in the list (resource of N-1 version) then it is not deleted.
cleanupUnusedResources iterates over all resource that do not belong to the list returned bu getAllResources method

Reply:
If an N-1 resource no longer exists in version N but it is still owned by the CDI CR the resource should get deleted when the CDI CR is deleted.

Reply:
Agree, resources with owner references will be cleaned up when the owner is removed.